### DISCUSSION TITLE
Are there two ways use btrfs?

### DISCUSSION THREAD

**Discussion by ningmingxiao** [moby/moby#47506] [General]
Since kernel 4.15 the btrfs filesystem can be used as lower filesystem for overlayfs (supporting the rename modes of exchange and whiteout).

1.Use the BTRFS storage driver
ref:https://docs.docker.com/storage/storagedriver/btrfs-driver/
2.Use BTRFS as as lower filesystem for overlayfs
ref:https://btrfs.readthedocs.io/en/latest/Interoperability.html#overlayfs
What are their differences? how to chose

**Comment by neersighted** [comment]
This is a duplicate of https://github.com/moby/moby/discussions/47544
